pandas/core/arrays/arrow/array.py

```diff
@@ -218,6 +218,9 @@ def _from_sequence(cls, scalars, *, dtype: Dtype | None = None, copy: bool = Fal
         if isinstance(scalars, cls):
             scalars = scalars._data
         elif not isinstance(scalars, (pa.Array, pa.ChunkedArray)):
+            if copy and is_array_like(scalars):
+                # pa array should not get updated when numpy array is updated
+                scalars = scalars.copy()
```

Comment:
Is it worth specifying `np.array(scalars, copy=True)` manually? There might be an off chance the arraylike doesn't have a `copy` method?

Reply:
This would cast our nullables to object, so not ideal

Reply:
Ah gotcha. Would calling `copy.copy(...)` be safer then?

Reply:
We need deepcopy, but this should do the trick

Reply:
> There might be an off chance the arraylike doesn't have a copy method?

might make more sense to raise than guess if this occurs?

Reply:
No preference here, cc @mroeschke